lib/hsa/mcwamp_hsa.cpp

```diff
@@ -2708,14 +2708,28 @@ class HSADevice final : public KalmarDevice
             case hc::EF_AMDGPU_MACH_AMDGCN_GFX906 : triple.append("906"); break;
         }
 
-        const auto isa{get_isa_name_from_triple(std::move(triple))};
+        unsigned sram_ecc_enabled = reader.get_flags() & hc::EF_AMDGPU_SRAM_ECC;
+        unsigned xnack_enabled = reader.get_flags() & hc::EF_AMDGPU_XNACK;
+        if (xnack_enabled) triple.append("+xnack");
+        if (sram_ecc_enabled) triple.append("+sram-ecc");
 
         struct isa_comp_data {
             hsa_isa_t isa_type;
             bool is_compatible;
         } co_data;
 
-        status = hsa_isa_from_name(isa.c_str(), &co_data.isa_type);
+        status = hsa_isa_from_name(triple.c_str(), &co_data.isa_type);
+
+        if (status != HSA_STATUS_SUCCESS) {
```

Comment:
I don't quite see what the intent is here. I suspect that we do not want to support loading newer code objects on older RTs, and even if we do this will absolutely do the wrong thing, because you'd load the wrong thing. Furthermore, surely you don't want `else if`...

Reply:
It's a kludge.  I'm not happy with it, but unless there is some way for HCC to validate the version of the RT, I'm not sure what else we can do.  

And, the "else if" is intended.  The std::erase will erase from the pos to the end of the string.  And since "+xnack" is first, we don't need to erase anymore, regardless of whether sram-ecc is set or not.  Honestly, my opinion is that none of this code should be in HCC/HIP.  It should all be contained in ROCR.  We should just be passing an ELF file to ROCR and getting back the isa_type.

Reply:
You can do "not this", which swallows errors assuming that they can only be due to mismatched RTs, and lies about the nature of the code that's in a code object. What are you actually trying to solve with this? Have you verified that it does anything for the problem that you are trying to solve?

Reply:
ok, yeah I can change the "if (status != HSA_STATUS_SUCCESS)".  So, I've verified that this works with a RT that isn't expecting the feature flags.  I'm working on verifying it on a system with a RT that does expect the feature strings; which was the original problem.

Reply:
It cannot work on a RT that does not expect the feature flags, because at that point you are lying about what is in the code object. So at best you have an odd mismatch.

Reply:
True.  This code, to remove the feature strings, is intended to be temporary.  We cannot commit the necessary changes to HCC/HIP without a RT that expects them.  And we cannot commit changes to the RT without a HCC/HIP that gives them.

Reply:
So i've validated that this fix works with a RT expecting the feature string, as well as a RT not expecting the feature string.  My proposal is that (after I fix the top "if"), we use this until we get a RT in our build/test envs that expects the feature string, then we can remove the code to handle the case of a "old" RT.